### DISCUSSION TITLE
On windows, the symfony server does not terminate after a first server:stop.

### DISCUSSION THREAD

**Discussion by mdeherder** [symfony/symfony#44699] [Q&A]
Here is a standard sequence that is problematic on windows and that works correctly on Linux

> C:\tools\Cmder
> λ `cd C:\Symfony\apphome`
> 

> C:\Symfony\apphome (master -> origin)
> λ `symfony serve -d`
> 
>  [OK] Web server listening
>       The Web server is using PHP CGI 8.0.10
>       https://127.0.0.1:8000
>       https://apphome.wip
> 
> 
> Stream the logs via symfony.exe server:log

> C:\Symfony\apphome (master -> origin)
> λ `symfony server:status`
> Local Web Server
>     Listening on https://127.0.0.1:8000
>     The Web server is using PHP CGI 8.0.10
> 
> Local Domains
>     https://apphome.wip
> 
> Workers
>     PID 28288: C:\tools\php80\php-cgi.exe -b 62133 -d error_log=C:\Users\miche\.symfony\log\25af236e540a890e64010b0f626a973f0c5a5593.log
> 
> Environment Variables
>     None

> C:\Symfony\apphome (master -> origin)
> λ `symfony server:stop`
> Stopping PHP-CGI
> Stopping Web Server
> 
> 
>  [OK] Stopped 2 process(es) successfully
> 

> C:\Symfony\apphome (master -> origin)
> λ `symfony server:status`
> Local Web Server
>     Listening on https://127.0.0.1:8000
>     The Web server is using PHP CGI 8.0.10
> 
> Local Domains
>     https://apphome.wip
> 
> Workers
>     PID 28288: C:\tools\php80\php-cgi.exe -b 62133 -d error_log=C:\Users\miche\.symfony\log\25af236e540a890e64010b0f626a973f0c5a5593.log
> 
> Environment Variables
>     None

> C:\Symfony\apphome (master -> origin)
> λ `symfony server:stop`
> Stopping PHP-CGI
> Stopping Web Server
> 
> 
>   TerminateProcess: Access is denied.

> C:\Symfony\apphome (master -> origin)
> λ `symfony server:status`
> Local Web Server
>     Not Running
> 
> Workers
>     No Workers
> 
> Environment Variables
>     None

**Comment by nayesoum** [comment]
bonjour, je rencontre le meme probleme vous avez trouver une solution?

**Comment by red-smeg** [comment]
I am getting the same problem on windows 11 running multiple server instances in powershell one per tab

**Comment by Dwza** [comment]
Same issue under windwos 10
The Web server is using PHP CGI 8.3.2
Symfony CLI version 5.8.6 (c) 2021-2024

**Comment by waldifubu** [comment]
Same problem. HAve always to delete pid file in .symfony/cache folder.
Before I guess it deleted the file automatically when was shutting down

**Comment by Dwza** [comment]
I guess if you close all programms that could have a handle on the server (eg console or ide with a terminal or so) than it seems like the handler on the server gets lost and it shuts down... at least it happend twice to me but it doesnt seem presistent :D

**Comment by Cartman34** [comment]
I got the same issue while I was using symfony server:start -d, so now, I am doing by another way...
start script:
```
symfony server:start > /dev/null 2>&1 &
echo $! > symfony_server.pid
```

stop script:
```
echo "Stopping Symfony server..."
symfony server:stop
sleep 1
if symfony server:status | grep -q "Not Running"; then
    echo "Symfony server stopped."
else
    echo "Symfony server did not stop. Try to kill process..."
    if [ -f symfony_server.pid ]; then
        TASK_PID=$(cat symfony_server.pid)
        kill $TASK_PID
        sleep 1
        rm symfony_server.pid
        if symfony server:status | grep -q "Not Running"; then
            echo "Killed Symfony server successfully."
        else
            echo "Failed to stop Symfony server."
        fi
    else
        echo "Unable to kill process."
    fi
fi
```

**Comment by issatengen** [comment]
I **closed** all the programs and re started the PC and it worked

**Comment by waldifubu** [comment]
ALternative:
If it's run in foreground I exit with CTRL + C
When I want to start again, I type in "bin/console server:stop" . Then the regarding files are also deleted and I can properly restart without issues.
